src/dc_imex.rs

```diff
@@ -490,7 +490,8 @@ pub unsafe fn dc_decrypt_setup_file(
             if let Some(plain) =
                 dc_pgp_symm_decrypt(passphrase, binary as *const libc::c_void, binary_bytes)
             {
-                payload = strdup(CString::new(plain).unwrap().as_ptr());
+                let payload_c = CString::new(plain).unwrap();
+                payload = strdup(payload_c.as_ptr());
```

Comment:
I think `dc_tools::to_cstring` has already these semantics and is what we use in other places

Reply:
`to_cstring` allocates memory with `malloc` and it is unsafe.

With `CString`, memory is only allocated by Rust allocator and all unsafety is contained in the functions that dereference the pointer. In particular, `as_ptr` is safe as long as you don't dereference the pointer it returns.

Reply:
We call `strdup` here anyway, so maybe it is better to use `dc_strdup` (or its wrapper `to_cstring`). In any case, there are some more similar CString/strdup snippets in this file, they all need to be cleaned up. I just fixed the use-after-free here.

Reply:
Yeah, I meant in combination with the `strdup`.